### PR TITLE
Dev generic

### PR DESCRIPTION
**Pull request by warquys**
Creating a draft for customer items.
It will be necessary to implement the old patches and events, to define the behavior of the items, possibly a behavior per type of item.
An additional patch is necessary to ensure the correct connection between the passage from a pickup to an item and vice versa.

Why a list of pickup and item and not just a the sérialisable ? 
> Because it's easier to manage this way

### CODE REVIEW

File: Exiled.CustomModules/API/Enums/UUCustomItem.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+// -----------------------------------------------------------------------
+
+namespace Exiled.CustomModules.API.Enums
+{
+    using Exiled.API.Features.Core.Generic;
+
+    /// <summary>
+    /// Represents the base enum class for all available custom item.
+    /// </summary>
+    public class UUCustomItem : UniqueUnmanagedEnumClass<uint, UUCustomTeamType>
```
Comment (NaoUnderscore) [inline Exiled.CustomModules/API/Enums/UUCustomItem.cs:15]: dafaq is going on here lmao

File: Exiled.CustomModules/API/Features/CustomItems/ItemBehaviour.cs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    /// <summary>
+    /// A tool to easily handle the custom item's logic.
+    /// </summary>
+    public abstract class ItemBehaviour : EPlayerBehaviour, IAdditiveSettings<ItemSettings>
+    {
+        /// <inheritdoc/>
+        public ItemSettings Settings { get; set; }
+
+        /// <summary>
+        /// Gets the role's configs.
```
Comment (NaoUnderscore) [inline Exiled.CustomModules/API/Features/CustomItems/ItemBehaviour.cs:25]:
```suggestion
        /// Gets the item's configs.
```

File: Exiled.CustomModules/API/Features/CustomItems/ItemBehaviour.cs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    using System.Reflection;
+
+    using Exiled.API.Features.Core;
+    using Exiled.API.Features.Core.Interfaces;
+    using Exiled.Events.EventArgs.Player;
+
+    /// <summary>
+    /// A tool to easily handle the custom item's logic.
+    /// </summary>
+    public abstract class ItemBehaviour : EPlayerBehaviour, IAdditiveSettings<ItemSettings>
```
Comment (NaoUnderscore) [inline Exiled.CustomModules/API/Features/CustomItems/ItemBehaviour.cs:19]:
```suggestion
    public abstract class ItemBehaviour : EItemBehaviour, IAdditiveSettings<ItemSettings>
```

File: Exiled.CustomModules/API/Features/CustomItems/CustomItem.cs [hunk 4]
```diff
@@ -0,0 +1,651 @@
...
+
+        /// <summary>
+        /// Gets all items belonging to a <see cref="CustomItem"/>.
+        /// </summary>
+        public static HashSet<Item> CustomItems => ItemManager.Keys.ToHashSet();
+
+        /// <summary>
+        /// Gets the <see cref="CustomItem"/>'s <see cref="Type"/>.
+        /// </summary>
+        public virtual Type BehaviourComponent { get; }
```
Comment (NaoUnderscore) [inline Exiled.CustomModules/API/Features/CustomItems/CustomItem.cs:64]:
```suggestion
        public abstract Type BehaviourComponent { get; }
```

File: Exiled.CustomModules/API/Features/CustomItems/CustomItem.cs [hunk 5]
```diff
@@ -0,0 +1,651 @@
...
+
+        /// <summary>
+        /// Gets all pickups belonging to a <see cref="CustomItem"/>.
+        /// </summary>
+        public static HashSet<Pickup> CustomPickups => PickupManager.Keys.ToHashSet();
+
+        /// <summary>
+        /// Gets all items belonging to a <see cref="CustomItem"/>.
+        /// </summary>
+        public static HashSet<Item> CustomItems => ItemManager.Keys.ToHashSet();
```
Comment (NaoUnderscore) [inline Exiled.CustomModules/API/Features/CustomItems/CustomItem.cs:59]: We may have to find another name for that.

**Review by NaoUnderscore — CHANGES_REQUESTED** [review]

File: Exiled.CustomModules/API/Features/CustomItems/CustomItem.cs (hunk 5 above)
Comment (warquys) [reply Exiled.CustomModules/API/Features/CustomItems/CustomItem.cs]: Check the new name

File: Exiled.CustomModules/API/Enums/UUCustomItem.cs (hunk 1 above)
Comment (warquys) [reply Exiled.CustomModules/API/Enums/UUCustomItem.cs]: Just a test, i forget to remove it

**Pull request merged**